web/packages/teleport/src/Console/DocumentKubeExec/DocumentKubeExec.story.test.tsx

```diff
@@ -0,0 +1,50 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import 'jest-canvas-mock';
+import { render } from 'design/utils/testing';
+
+import {
+  DocumentKubeExecWrapper,
+  getContexts,
+  baseDoc,
+} from './DocumentKubeExec.story';
+
+test('render DocumentKubeExec', async () => {
+  const { ctx, consoleCtx } = getContexts();
+
+  Object.defineProperty(window, 'matchMedia', {
+    writable: true,
+    value: jest.fn().mockImplementation(query => ({
+      matches: false,
+      media: query,
+      onchange: null,
+      addListener: jest.fn(), // Deprecated
+      removeListener: jest.fn(), // Deprecated
+      addEventListener: jest.fn(),
+      removeEventListener: jest.fn(),
+      dispatchEvent: jest.fn(),
+    })),
+  });
+
+  const { container } = render(
+    <DocumentKubeExecWrapper ctx={ctx} consoleCtx={consoleCtx} doc={baseDoc} />
+  );
+  expect(container.firstChild).toMatchSnapshot();
```

Comment:
I don't view this strictly as a requirement since I don't think you usually work on frontend stuff, but if you have time then I'd timebox this to like 30–60 minutes and see how far you can get with the idea below, it'd help us out in the long term! If not then I guess the snapshot test is fine.

---

I understand that this was mostly copied from `DocumentSsh`, but it'd be nice to add a regular test instead of a snapshot test, as we generally want to reduce the number of snapshot tests.

It could boil down to duplicating the setup from the story (so that the test does not depend on the story) and then performing a simple check. It could be just writing to the tty and then checking if that gets rendered on the screen.

If all else fails, the check could end up as `data-testid="terminal"` added to the div rendered by `Terminal`:

https://github.com/gravitational/teleport/blob/43d050b82158773fd20d23223308d80be5160e3d/web/packages/teleport/src/Console/DocumentSsh/Terminal/Terminal.tsx#L169-L177

…and then checking if the div got rendered on the screen:

https://github.com/gravitational/teleport/blob/43d050b82158773fd20d23223308d80be5160e3d/web/packages/teleport/src/components/ToolTipNoPermBadge/ToolTipNoPermBadge.test.tsx#L37

(Though using `data-testid` should be the last resort.)